File: JDApp-Java/home/src/main/java/com/jd/home/application/HomeApplication.java
package com.jd.home.application;

import android.app.Application;
import android.util.Log;

import com.renny.mylibrary.IAppInit;

/*
    @see https://github.com/ren93/initiator
 */

public class HomeApplication implements IAppInit {
    @Override
    public void init(Application application) {
        Log.d("init==", "HomeApplication");
    }
}

File: JDApp-Kotlin/home/src/main/java/com/jd/home/application/HomeApplication.kt
package com.jd.home.application

import android.app.Application
import android.content.Context
import android.util.Log
import com.google.auto.service.AutoService
import com.jd.core.base.IApplication

@AutoService(IApplication::class)
class HomeApplication : IApplication {
    override fun onAttachBaseContext(context: Context) {
    }

    override fun onCreate(application: Application) {
        Log.d("init==", "HomeApplication")
    }

    override fun onAsyncInit() {

    }


    override fun onTerminate(application: Application) {
    }


}
